cmd/minishift/cmd/profile/profile_delete.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profile
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/docker/machine/libmachine"
+	cmdUtil "github.com/minishift/minishift/cmd/minishift/cmd/util"
+	"github.com/minishift/minishift/pkg/minikube/cluster"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	profileActions "github.com/minishift/minishift/pkg/minishift/profile"
+	pkgUtil "github.com/minishift/minishift/pkg/util"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+)
+
+var (
+	profileDeleteCmd = &cobra.Command{
+		Use:   "delete PROFILE_NAME",
+		Short: "delete profiles.",
+		Long:  "deletes an existing profile.",
+		Run:   runProfileDelete,
+	}
+	force bool
+)
+
+func runProfileDelete(cmd *cobra.Command, args []string) {
+	if len(args) == 0 {
+		atexit.ExitWithMessage(1, emptyProfileError)
+	} else if len(args) > 1 {
+		atexit.ExitWithMessage(1, extraArgumentError)
+	}
+
+	profileName := args[0]
+	var defultProfile bool
+
+	if !cmdUtil.IsValidProfile(profileName) {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error: '%s' is not a valid profile", profileName))
+	}
+
+	profileActions.UpdateMiniConstants(profileName)
```

Comment:
What are `MiniConstants`? This name is not descriptive as it is related to profile, and not just ('a') constant(s). Aren't they paths? `UpdatePathsRelatedToProfile(profileName)` or something similar.